domain/operation/service/service.go

```diff
@@ -84,6 +84,11 @@ type State interface {
 	// query function which returns the list of task ids for the given machine.
 	InitialWatchStatementMachineTask() (string, string)
 
+	// StartTask sets the task start time and updates the status to running.
+	// Returns [operationerrors.TaskNotFound] if the task does not exist,
+	// and [operationerrors.TaskNotPending] if the task is not pending.
+	StartTask(ctx context.Context, taskID string) error
```

Comment:
Should this be called `BeginTask()` maybe? We already have `Start*`: https://github.com/juju/juju/blob/5e8f5af2c9f1b33b3f05134ef593a07826acca63/domain/operation/service/start.go#L13-L32

Reply:
@gfouillet @nvinuesa, In fact, the opposite is true. The methods provided should be renamed to `AddExecOperation` etc. They are setting up for tasks to be run, but not actually doing it in those methods.

`StartTask` indicates that the machine or unit is actually running the task.